### PR TITLE
Fix Windows problem not finding ASIO.

### PR DESCRIPTION
**Pull request by clalancette**
It basically needs an upgrade to setup-ros and action-ros-ci.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

This still doesn't totally fix Windows, but gets us another step closer.

### CODE REVIEW

**Comment by clalancette** [comment]
@jacobperron FYI.

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -11,10 +11,10 @@ jobs:
       run: |
         sudo apt update -qq
         sudo apt install -y default-jdk
-    - uses: ros-tooling/setup-ros@3b3f060
+    - uses: ros-tooling/setup-ros@b1c8132
```
Comment (jacobperron) [inline .github/workflows/build_and_test.yml:14]: Looks like we can use the release tag [0.0.24](https://github.com/ros-tooling/setup-ros/tree/0.0.24).

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -28,11 +28,12 @@ jobs:
         java-version: '11.0.6' # The JDK version to make available on the path.
         java-package: jdk
         architecture: x64
-    - uses: ros-tooling/setup-ros@3b3f060
+    - uses: ros-tooling/setup-ros@b1c8132
```
Comment (jacobperron) [inline .github/workflows/build_and_test.yml:31]: Looks like we can use the release tag [0.0.24](https://github.com/ros-tooling/setup-ros/tree/0.0.24).

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (clalancette) [reply .github/workflows/build_and_test.yml]: Fixed in 406844e.

File: .github/workflows/build_and_test.yml (hunk 2 above)
Comment (clalancette) [reply .github/workflows/build_and_test.yml]: Fixed in 406844e.

**Pull request merged**